### PR TITLE
Tide shard config manager for ocplifecycle events with feature freeze

### PR DESCRIPTION
**Pull request by jmguzik**
Prototype with feature freeze implemented

/cc @droslean @danilo-gemoli @bear-redhat 

@droslean is currently absent so @danilo-gemoli and @bear-redhat please look instead :) 

Signed-off-by: Jakub Guzik <jguzik@redhat.com>

### CODE REVIEW

File: cmd/branchingconfigmanagers/tide-config-manager/main.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+func (ffe *featureFreezeEvent) ensureFeatureFreezeBugs(q *prowconfig.TideQuery) {
+	requiredLabels := sets.NewString(q.Labels...)
+	branches := sets.NewString(q.IncludedBranches...)
+	if branches.Intersection(ffe.mainMaster).Len() == 0 {
+		return
+	}
+	if requiredLabels.Intersection(ffe.excludedLabels).Len() > 0 {
+		return
+	}
+	requiredLabels = requiredLabels.Union(sets.NewString(validBug))
```
Comment (danilo-gemoli) [inline cmd/branchingconfigmanagers/tide-config-manager/main.go:167]: Should we check if `validBug` is already contained?

Comment (danilo-gemoli) [reply cmd/branchingconfigmanagers/tide-config-manager/main.go:167]: Ok it is not needed, it is just a set.

File: cmd/branchingconfigmanagers/tide-config-manager/main.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+func (ffe featureFreezeEvent) GetDataFromProwConfig(pc *prowconfig.ProwConfig) {
+	for _, query := range pc.Tide.Queries {
+		branches := sets.NewString(query.IncludedBranches...)
+		labels := sets.NewString(query.Labels...)
+		ffe.rewriteReposToExcludeOrgs(&query)
+		if branches.Intersection(ffe.openshiftReleaseBranchesPlus1).Len() > 0 && labels.Has(staffEngApproved) {
+			ffe.repos.Insert(query.Repos...)
+			continue
+		}
+		if branches.Intersection(ffe.openshiftReleaseBranchesPlus1).Len() > 0 && labels.Has(staffEngApproved) {
```
[the remark this answers is not in the document]
Comment (jmguzik) [reply cmd/branchingconfigmanagers/tide-config-manager/main.go:135]: nice catch! It won't affect the run as there is continue but 2nd if is completely useless.

**Pull request merged**